Source/WebCore/animation/AnimationEffect.cpp

```diff
@@ -190,6 +190,18 @@ ExceptionOr<void> AnimationEffect::updateTiming(Document& document, std::optiona
         }
     }
 
+    if (auto iterations = timing->iterations) {
+        // https://github.com/w3c/csswg-drafts/issues/11343
+        if (std::isinf(*iterations)) {
+            if (RefPtr animation = m_animation.get()) {
+                if (RefPtr timeline = animation->timeline()) {
+                    if (timeline->isProgressBased())
+                        return Exception { ExceptionCode::TypeError };
```

Comment:
Can you add an error message? Otherwise, web developers will just see "TypeError" with no other info in the console.

Ditto for the remaining exceptions in this method (although they can be done separately)

```suggestion
                        return Exception { ExceptionCode::TypeError, "The number of iterations cannot be set to Infinity for progress-based animations"_s  };
```